### PR TITLE
Add Jaro-Winkler similarity UDF

### PR DESCRIPTION
**Pull request by bnaul**
Already using it in Patterns but I found that the implementation was giving values that didn't match DuckDB's, which we also use. This version matched more closely and I thought it should get committed somewhere.

Pulled from https://github.com/NaturalNode/natural/blob/19c57d1cbc7310370eaf8c44379ea37df943af0c/lib/natural/distance/jaro-winkler_distance.js#L105

### CODE REVIEW

File: functions/string/sql/jaro_winkler_similarity.sql [hunk 1]
```diff
@@ -0,0 +1,86 @@
+CREATE OR REPLACE FUNCTION bqutils.string.jaro_winkler_similarity(s1 STRING, s2 STRING)
+RETURNS FLOAT64
+LANGUAGE js AS """
+// Adapted from https://github.com/NaturalNode/natural/blob/19c57d1cbc7310370eaf8c44379ea37df943af0c/lib/natural/distance/jaro-winkler_distance.js#L105
+// Computes the Jaro distance between two strings
+function distance(s1, s2) {
+  if (typeof(s1) !== 'string' || typeof(s2) !== 'string') {
```
Comment (ajnadel) [inline functions/string/sql/jaro_winkler_similarity.sql]: Is this behavior (returning 0 for out-of-domain inputs) normal for BQ UDFs as opposed to erroring?

File: functions/string/sql/jaro_winkler_similarity.sql [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      matches1[i] = true;
+      matches2[k] = true;
+      m++;
+      break;
+    }
+  }
+
+  // nothing matched
+  if (m === 0) {
+    return 0.0;
```
Comment (ajnadel) [inline functions/string/sql/jaro_winkler_similarity.sql:41]: nit: make this 0 for consistency with other `number` literals in this func?

**Review by ajnadel — APPROVED** [review]

File: functions/string/sql/jaro_winkler_similarity.sql (hunk 1 above)
Comment (bnaul) [reply functions/string/sql/jaro_winkler_similarity.sql]: the only case we'd hit it is NULL, I changed to handle that more explicitly

File: functions/string/sql/jaro_winkler_similarity.sql (hunk 2 above)
Comment (bnaul) [reply functions/string/sql/jaro_winkler_similarity.sql:41]: I think I'm just gonna leave it so we exactly match the source function that I copy pasted

**Pull request merged**